subsys/bluetooth/host/smp.c

```diff
@@ -1632,7 +1632,15 @@ static void smp_pairing_complete(struct bt_smp *smp, uint8_t status)
 {
 	struct bt_conn *conn = smp->chan.chan.conn;
 
-	LOG_DBG("status 0x%x", status);
+	LOG_DBG("got status 0x%x", status);
+
+	if (conn->state != BT_CONN_CONNECTED) {
+		/* If disconnection has been triggered in between the security update
+		 * and the call to this function we need to abort the pairing.
+		 */
+		LOG_WRN("Not connected!");
+		status = BT_SMP_ERR_UNSPECIFIED;
+	}
```

Comment:
Correct me if I am wrong, but by setting `status = BT_SMP_ERR_UNSPECIFIED` would correctly call the callbacks, but what is stopping it from calling `bt_smp_start_security` at the end of this function? 

```c
	if (conn->sec_level != conn->required_sec_level) {
		bt_smp_start_security(conn);
}
```

Would it be possible to add a test case for this scenario?

Reply:
I think you're right. I did not find any place where we clear the field. Should we not do it at disconnection in principle?